### DISCUSSION TITLE
Command line

### DISCUSSION THREAD

**Discussion by jerone** [DevToys-app/DevToys#1428] [Q&A] [answered]
Two questions:

1. I read somewhere that the previous version (v1) was able to open on a particular tool, via `devtoys:?tool={tool name}`. Does that work in v2?

2. Is the CLI suposted to work in every terminal and from every directory, like shown on the homepage screenshot? Running the same command as in the screenshot throws not found errors.
  ![image](https://github.com/user-attachments/assets/3bcf20c2-551b-48ad-8da3-ba27c0aac916)

**Comment by veler** [comment, marked as answer]
Hello !

1. It is supported, but not well documented I admit. You can do it using a command line like `DevToys.exe --tool:HashAndChecksumGenerator`. There's currently no way to add extra parameters to fill in the inputs, although it should be achievable by sending the extra data to the Smart Detection service, which currently just read the content of the clipboard.

2. On Linux, if you install `DevToys.CLI` using the DEB package, you should be able to invoke it from anywhere like on the screenshot. This isn't the case on Windows and MacOS at the moment. You need to use the absolute path to the DevToys.CLI executable.

**Reply by jerone** [reply]
Thank you for the explanation.

**Comment by davidegiacometti** [comment]
Hey @veler, thanks for clarifying that!
I have looked for DevToys 2.0 deep-linking support in the past without success.

@jerone linked this discussion in the repo of a DevToys 1.0 plugin I made for PowerToys Run: https://github.com/davidegiacometti/PowerToys-Run-DevToys/issues/1

Hard coding the utilities isn't the best solution but it's been stable enough for DevToys 1.0 in the past 😄 
Some user have asked for 2.0 support and would like to try implementing it.

That said, I am aware of different installers available: MSIX, classic and portable.
I am using MSIX from Store and I am not able to run `DevToys.exe` from `C:\Program Files\WindowsApps` (I think this is expected).
But, I should be able to launch it with
`shell:AppsFolder\64360VelerSoftware.DevToys-Preview_j80j2txgjg9dj!App --tool:HashAndChecksumGenerator`

`shell:AppsFolder\64360VelerSoftware.DevToys_j80j2txgjg9dj!App` opens DevToys 1.0

Any idea? Am I missing something here?
Thanks!
